scripts/dotnet-cli-build/CompileTargets.cs

```diff
@@ -52,12 +74,37 @@ public static BuildTargetResult UpdateBuild(BuildTargetContext c)
 
         // Moving PrepareTargets.RestorePackages after PackagePkgProjects because managed code depends on the
         // Microsoft.NETCore.App package that is created during PackagePkgProjects.
-        [Target(nameof(PrepareTargets.Init), nameof(CompileCoreHost), nameof(PackagePkgProjects), nameof(PrepareTargets.RestorePackages), nameof(CompileStage1), nameof(CompileStage2))]
+        [Target(nameof(PrepareTargets.Init), nameof(CompileCoreHost), nameof(PackagePkgProjects), nameof(CompileTargets.GenerateStubHostPackages), nameof(PrepareTargets.RestorePackages), nameof(CompileStage1), nameof(CompileStage2))]
         public static BuildTargetResult Compile(BuildTargetContext c)
         {
             return c.Success();
         }
 
+        [Target]
+        public static BuildTargetResult GenerateStubHostPackages(BuildTargetContext c)
+        {
+            string currentRid = GetRuntimeId();
+            var buildVersion = c.BuildContext.Get<BuildVersion>("BuildVersion");
+
+            foreach (var hostPackageId in HostPackages)
+            {
+                foreach (var rid in NetCoreAppRids)
+                {
+                    if (! rid.Equals(currentRid))
+
+                    {
+                        CreateDummyRuntimeNuGetPackage(
```

Comment:
> ultimately I think we need to add back the --runtimes switch to dotnet restore, do you know if that is more or less work then doing this stub generation and dealing with that?

@yishaigalatzer @piotrpMSFT @davidfowl and I all had this conversation last week.

@yishaigalatzer's response was "it is a very easy task to add back in --runtimes" but he wants us (.NET and ASP.NET) to come to consensus on this topic, instead of putting it in and taking it out, and putting it back in.
There were also concerns that no other customer would need this, and that this is a CLI-only issue.